docs/src/main/asciidoc/chapters/iterator_design.txt

```diff
@@ -145,8 +146,16 @@ alter the internal state of the Iterator.
 
 These methods simply return the current Key-Value pair for this iterator. If `hasTop` returns true,
 both of these methods should return non-null objects. If `hasTop` returns false, it is undefined
-what these methods should return. Multiple calls to these methods should not alter the state
-of the Iterator like `hasTop`.
+what these methods should return. Like `hasTop`, multiple calls to these methods should not alter 
+the state of the Iterator.
+
+When saving a Key or Value from a source iterator's `getTopKey` or `getTopValue` methods
+for use after calling `next` on the source iterator (e.g., when cacheing keys or values
+from the source iterator), it is important to copy the Key or Value into a new object 
+because the source iterator may reuse the Key or Value objects for performance reasons.
```

Comment:
Actually, I reread your additions to the SKVI javadoc and I like the way you worded things there. If you can, a/some examples of when/how this might happen would be good if you can share. I know I've done things that would be not recommended by your warning (e.g. collapse a subset of columns in a row into a single K/V) without running into issues.